_posts/2020-03-24-dask-developer-workshop.markdown

```diff
@@ -0,0 +1,85 @@
+---
+layout: single
+title:  "Tales from D.C.: Dask Developer Workshop 2020"
+date:   2020-03-24 20:00:00 +0100
+tags: technology python data-engineering dask distributed
+header:
+  overlay_image: assets/images/tech_gear_banner.jpg
+  overlay_filter: 0.2
+  show_overlay_excerpt: false
+author: Lucas Rademaker
+author_profile: true
+---
+# Tales from D.C.: Dask Developer Workshop 2020
+
+## Setting
+Back in January this year, [Dask](https://dask.org/) community members received an invite for a workshop in Washington DC with the goal of "connecting advanced users with developers".
+
+## The workshop
+
+The schedule of the workshop alone was quite satisfying to me: a number of talks, involving users and developers of Dask and related projects (e.g. NumPy, xarray, Arrow), combined with working sessions, wherein one had the opportunity to discuss or code with other attendees.
+
+What follows is an outline of my experience at the workshop.
+### Talks
+##### ETL Pipelines
+ - Blue Yonder and [Kartothek](https://github.com/JDASoftwareGroup/kartothek).
+ 
+    My colleague Florian Jetter was the first to present. He outlined our use case of executing Dask pipelines coupled with tight SLA's (Service Level Agreements), which obliges us to make sure that our Dask pipelines finish successfully each run, as usually we do not have time to re-schedule. He also talked about some of the issues we had faced in the past and others which we are facing in the present (mostly related to stability).
+
+- Prefect
+
+    TODO
+
+##### Deployment
+  - dask-gateway
+
+    TODO
+
+  - dask-jobqueue
+
+    TODO
+
+##### Other kids on the block
+ - xarray
+ 
+    TODO
+
+ - RAPIDS
+ 
+    TODO
+
+ - NumPy
+ 
+    TODO
+
+ - dask-ML
+ 
+    TODO
+
+### Dask issues and potential improvements (FIXME)
+
+It was nice to see that we were not the only ones facing these kind of issues. In fact, almost all major issues mentioned at the workshop were shared among multiple attendees. Some of these were:
+- Stability: in particular network stability, lack of replication of results, downscaling issues.
+- Performance: lack of memory backpressure (OOME's are a thing), other issues involving suboptimal graph optimization  (e.g. task fusing).
+- Suboptimal work stealing.
+
+Other issues mentioned involved indexing/partitioning, the lack of a multi-index and "non-functional" items such as documentation and examples.
+
+
+The atmosphere of the workshop was great, people were eager to share their experiences with Dask and the ecosystem surrounding it. This provided for some interesting discussions.
+
+
+### Working sessions (FIXME)
+
+Note: the following is purely a result of personal experience, I'm sure there are plenty of interesting things were done or talked about of which I am not aware.
```

Comment:
I would probably leave this sentence out completely, as there is no information gain. If you want to introduce the topics you specifically discussed during the workshop, then just say it like that:
"During the workshop, I specifically worked on blahblahblah and did XYZ with Mrs. FamousDeveloper ..."